presto-product-tests/src/test/java/com/facebook/presto/tests/hive/TestTablePartitioningSelect.java

```diff
@@ -104,6 +104,23 @@ public void testSelectPartitionedHiveTableDifferentFormats()
         assertProcessedLinesCountEquals(selectFromOnePartitionsSql, onePartitionQueryResult, 1);
     }
 
+    @Test(groups = HIVE_CONNECTOR)
+    public void testUnionPartitionedHiveTableDifferentFormats()
+            throws SQLException
+    {
+        String tableNameInDatabase = tablesState.get(TABLE_NAME).getNameInDatabase();
+
+        String selectFromAllPartitionsSql = "SELECT * FROM " + tableNameInDatabase + " UNION ALL SELECT * FROM " + tableNameInDatabase;
+        QueryResult allPartitionsQueryResult = query(selectFromAllPartitionsSql);
+        assertThat(allPartitionsQueryResult).containsOnly(row(42, 1), row(42, 2), row(42, 1), row(42, 2));
+        assertProcessedLinesCountEquals(selectFromAllPartitionsSql, allPartitionsQueryResult, 4);
```

Comment:
I'm not sure..  the other partition tests do this; I think it's something to do with making sure that the correct number of rows are processed across all of the partitions. @sogorkis can you clarify?